Tensile/Source/client/main.cpp

```diff
@@ -599,31 +691,36 @@ int main(int argc, const char* argv[])
                     {
                         while(listeners.needMoreRunsInSolution())
                         {
-                            auto inputs = dataInit->prepareGPUInputs(problem);
-
-                            auto kernels = solution->solve(problem, *inputs, *hardware);
+                            std::vector<std::shared_ptr<ContractionInputs>> inputs;
+                            std::vector<std::vector<KernelInvocation>>      kernels;
+                            for(size_t i = 0; i < flush_count; i++)
+                            {
+                                inputs.push_back(dataInit[i]->prepareGPUInputs(problem));
+                                kernels.push_back(
+                                    solution->solve(problem, *(inputs[i]), *hardware));
+                            }
 
                             size_t       warmupInvocations = listeners.numWarmupRuns();
-                            size_t       eventCount        = gpuTimer ? kernels.size() : 0;
+                            size_t       eventCount        = gpuTimer ? kernels[0].size() : 0;
                             TimingEvents warmupStartEvents(warmupInvocations, eventCount);
                             TimingEvents warmupStopEvents(warmupInvocations, eventCount);
 
                             for(int i = 0; i < warmupInvocations; i++)
                             {
                                 listeners.preWarmup();
                                 if(gpuTimer)
-                                    HIP_CHECK_EXC(adapter.launchKernels(kernels,
+                                    HIP_CHECK_EXC(adapter.launchKernels(kernels[0],
                                                                         stream,
                                                                         warmupStartEvents[i],
                                                                         warmupStopEvents[i]));
                                 else
-                                    HIP_CHECK_EXC(
-                                        adapter.launchKernels(kernels, stream, nullptr, nullptr));
+                                    HIP_CHECK_EXC(adapter.launchKernels(
+                                        kernels[0], stream, nullptr, nullptr));
```

Comment:
Warmup is only done with the first buffer.
1) Is this how it is done in rocblas-bench?
2) Does this affect warmup or timed runs in any way?
3) Will the first timed run use the same buffer as the warmup runs?

Reply:
1. Yes rocblas also only uses first buffer for warmups
2/3. The first timed run will not use the same buffer as warmup, which is the intended behaviour to avoid using cached memory.